75 devolop events view page display personalized greeting using firebase name

SufyanCS opened the pull request:
## **Describe what does this pull request do?**

 add a personalized message to greet the user with the first name when they log in to the website
![image](https://user-images.githubusercontent.com/119085953/233919350-8a9c5c4c-6f49-4dfe-bb18-d1078f0de41a.png)

SufyanCS committed:
feat(develop personalized greeting message): add personalized message to greet the user

add a personalized message to greet the user with the first name when they log in to the website

SufyanCS committed:
Merge branch 'develop' of github.com:202212-GIZ-YE-FEW/Web_Wizards_Pebble into 75-devolop-events-view-page-display-personalized-greeting-using-firebase-name

SufyanCS committed:
fix(personalized greeting message): fix personalized greeting message while refreshing the page

fixes a bug with the personalized greeting message feature that occurs when the user refreshes the
page. Previously, the message would disappear when the user refreshed the page, but now it persists
across page refreshes.

iiAbady commented on src/pages/events/index.js:
```diff
@@ -7,13 +7,14 @@ import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 // React imports
 import React, { useEffect, useState } from "react";
 
+import { auth } from "@/lib/useAuth";
```
Great work @SufyanCS! To follow the principle of `DRY` you will need to use the auth context without caring of doing `onAuthStateChanged` because all of that is being handled by the context.

So remove all authentication code and import the user from the context 
```javascript
import { useAuthContext } from '../../../context/AuthContext'
const { user }  = useAuthContext();
```

iiAbady commented on src/pages/events/index.js:
```diff
@@ -120,14 +121,22 @@ function Events() {
         }
     };
-
+    const [currentUser, setCurrentUser] = useState(
+        auth?.currentUser?.displayName
+    );
+    useEffect(() => {
+        const unsubscribe = auth.onAuthStateChanged((user) => {
+            setCurrentUser(user?.displayName.split(" ")[0]);
+        });
+        return unsubscribe;
+    }, []);
```
All of this is not needed!

iiAbady commented on src/pages/events/index.js:
```diff
@@ -120,14 +121,22 @@ function Events() {
                 <div className='flex items-start flex-col gap-3'>
                     <h1 className='text-6xl font-extrabold text-black-100'>
-                        {t("welcome")}, John!
+                        {t("welcome")}, {currentUser}
```
it should be `{user.displayName}`

SufyanCS committed:
refactor(refactor authentication context usage): update the usage of the useAuthContext hook

replace the useState and useEffect hooks with useAuthContext,To improve its clarity and
maintainability
